> @sikmir could you try set in the derivation this?:
> 
> ```nix
> GETTEXT_SYSTEM = true;
> ```
> 
> I'm trying to follow [this](https://github.com/Koka/gettext-rs#manual-configuration).

The same result:
```
..
   Compiling gettext-rs v0.5.0
     Running `rustc --crate-name gettextrs /private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/newsboat-2.21-vendor.tar.gz/gettext-rs/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -Cembed-bitcode=no -C metadata=5a6fb6ac89878a0e -C extra-filename=-5a6fb6ac89878a0e --out-dir /private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/source/target/release/deps -C linker=/nix/store/sw7b2gcklaj3iy5ikv7ag163bpry3ivb-clang-wrapper-7.1.0/bin/cc -L dependency=/private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/source/target/release/deps --extern gettext_sys=/private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/source/target/release/deps/libgettext_sys-00750fbde98dd8d4.rmeta --extern locale_config=/private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/source/target/release/deps/liblocale_config-daaaa1299af9fb6f.rmeta --cap-lints allow -L native=/private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/source/target/release/build/gettext-sys-d162d66149a94350/out/lib`
error: could not find native static library `intl`, perhaps an -L flag is missing?

error: aborting due to previous error

error: could not compile `gettext-sys`.

Caused by:
  process didn't exit successfully: `rustc --crate-name gettext_sys /private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/newsboat-2.21-vendor.tar.gz/gettext-sys/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -Cembed-bitcode=no --cfg 'feature="gettext-system"' -C metadata=00750fbde98dd8d4 -C extra-filename=-00750fbde98dd8d4 --out-dir /private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/source/target/release/deps -C linker=/nix/store/sw7b2gcklaj3iy5ikv7ag163bpry3ivb-clang-wrapper-7.1.0/bin/cc -L dependency=/private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/source/target/release/deps --cap-lints allow -L native=/private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-newsboat-2.21.drv-0/source/target/release/build/gettext-sys-d162d66149a94350/out/lib -l framework=CoreFoundation -l dylib=iconv -l static=intl` (exit code: 1)
warning: build failed, waiting for other jobs to finish...
error: build failed
make: *** [Makefile:108: target/release/libnewsboat.a] Error 101
builder for '/nix/store/bibz7681a2kfi4nmcn4gb494q68xil8z-newsboat-2.21.drv' failed with exit code 2
error: build of '/nix/store/bibz7681a2kfi4nmcn4gb494q68xil8z-newsboat-2.21.drv' failed
```

--- a/pkgs/applications/networking/feedreaders/newsboat/default.nix
+++ b/pkgs/applications/networking/feedreaders/newsboat/default.nix
@@ -1,26 +1,29 @@
 { stdenv, rustPlatform, fetchFromGitHub, stfl, sqlite, curl, gettext, pkg-config, libxml2, json_c, ncurses
-, asciidoctor, libiconv, Security, makeWrapper }:
+, asciidoctor, libiconv, Security, Foundation, makeWrapper }:
 
 rustPlatform.buildRustPackage rec {
   pname = "newsboat";
-  version = "2.20.1";
+  version = "2.21";
 
   src = fetchFromGitHub {
     owner = "newsboat";
     repo = "newsboat";
     rev = "r${version}";
-    sha256 = "1i9dpkdlsm3ya0w2x4c8kplrp3qzd8slbkcqvzfpqggb67gvczvv";
+    sha256 = "0ignfmh5193bigvk9f057r0r4yaxymxv2afycn2b98w05gljccb6";
   };
 
-  cargoSha256 = "1ykffx2lhn4w56qm1wypkg9wsqpvzzrz419qkl95w1384xf3f7ix";
+  cargoSha256 = "16652i2hbs6d3fam2hdlc947i5nrb3na186zfcb4nfh7hnb7lh8g";
 
   postPatch = ''
     substituteInPlace Makefile --replace "|| true" ""
-  '' + stdenv.lib.optionalString stdenv.isDarwin ''
-    # Allow other ncurses versions on Darwin
-    substituteInPlace config.sh \
-      --replace "ncurses5.4" "ncurses"
-  '';
+  ''
+    # TODO: Check if that's still needed
+    + stdenv.lib.optionalString stdenv.isDarwin ''
+      # Allow other ncurses versions on Darwin
+      substituteInPlace config.sh \
+        --replace "ncurses5.4" "ncurses"
+    ''
+  ;
 
   nativeBuildInputs = [
     pkg-config
@@ -29,7 +32,7 @@ rustPlatform.buildRustPackage rec {
   ] ++ stdenv.lib.optionals stdenv.isDarwin [ makeWrapper ncurses ];
 
   buildInputs = [ stfl sqlite curl libxml2 json_c ncurses ]
-    ++ stdenv.lib.optionals stdenv.isDarwin [ Security libiconv gettext ];
+    ++ stdenv.lib.optionals stdenv.isDarwin [ Security Foundation libiconv gettext ];
 
   postBuild = ''
     make prefix="$out"
@@ -37,6 +40,13 @@ rustPlatform.buildRustPackage rec {
 
   # TODO: Check if that's still needed
   NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.isDarwin " -Wno-error=format-security";
+
+  # https://github.com/NixOS/nixpkgs/pull/98471#issuecomment-703100014 . We set
+  # these for all platforms, since upstream's gettext crate behavior might
+  # change in the future.
+  GETTEXT_LIB_DIR = "${stdenv.lib.getLib gettext}/lib";
+  GETTEXT_INCLUDE_DIR = "${stdenv.lib.getDev gettext}/include";
+  GETTEXT_BIN_DIR = "${stdenv.lib.getBin gettext}/bin";
 
   doCheck = true;
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5745,7 +5745,7 @@ in
   nm-tray = libsForQt5.callPackage ../tools/networking/network-manager/tray.nix { };
 
   newsboat = callPackage ../applications/networking/feedreaders/newsboat {
-    inherit (darwin.apple_sdk.frameworks) Security;
+    inherit (darwin.apple_sdk.frameworks) Security Foundation;
   };
 
   grocy = callPackage ../servers/grocy { };

